datahugger/services.py

```diff
@@ -445,3 +451,113 @@ class B2shareDataset(DatasetDownloader):
     ATTR_SIZE_JSONPATH = "size"
     ATTR_HASH_JSONPATH = "checksum"
     ATTR_HASH_TYPE_VALUE = "md5"
+
+
+class YodaDataset(DatasetDownloader):
+    """Downloader for Yoda repositories."""
+
+    def _get_contents_url(self):
+        """Resolve the root folder of the contents of a Yoda data package."""
+        url_to_use = (
+            self.resource if isinstance(self.resource, str) else self.resource.resolve()
+        )
+        res = requests.get(url_to_use)
+        res.raise_for_status()
+        soup = BeautifulSoup(res.content, "html.parser")
+        contents_link = soup.find("a", string=re.compile(r"View contents"))
```

Comment:
Wouldn't it be better to select based on a class? The current class is `btn btn-primary access-btn view-contents`, which appears to be a suitable and stable option for navigating this URL (e.g. `view-contents`). We also don't need regexp.

Reply:
Right, matching on the class is a better option. I've updated it.